src/App.tsx

```diff
@@ -1,26 +1,216 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useRef, useState } from 'react';
+import * as todoService from './api/todos';
+import { USER_ID } from './api/todos';
+import { Todo } from './types/types';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+import { Notifications } from './components/Notifications/Notifications';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState('all');
+  const [inputTitle, setInputTitle] = useState('');
+  const myId = USER_ID;
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingId, setDeletingId] = useState<number | null>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const timeOutRef = useRef<NodeJS.Timeout | number>(0);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      if (loading) {
+        inputRef.current.disabled = true;
+      } else {
+        inputRef.current.disabled = false;
+        inputRef.current.focus();
+      }
+    }
+  }, [loading]);
+
+  const completedCount = todos.filter(todo => todo.completed).length;
+
+  function close() {
```

Comment:
It is important to follow the same way of creating functions everywhere, fix it everywhere
```suggestion
  const close = () => {
```